blocks/api/parser.js

```diff
@@ -181,6 +181,28 @@ export function createBlockWithFallback( name, innerHTML, attributes ) {
 		// as invalid, or future serialization attempt results in an error
 		block.originalContent = innerHTML;
 
+		// When a block is invalid, attempt to validate again using a supplied `deprecated` definition.
+		// This allows blocks to modify their attribute and markup structure without invalidating
+		// content written in previous formats.
+		if ( ! block.isValid && blockType.deprecated ) {
+			let attributesParsedWithDeprecatedVersion;
+			const hasValidOlderVersion = find( blockType.deprecated, ( oldBlockType ) => {
```

Comment:
[`Array#some`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some) (or Lodash's equivalent) would be a more appropriate method here, since we don't care about the found value, just that it exists.